tests/cpp/include/test_ndarray_utils.h

```diff
@@ -47,16 +49,13 @@ inline void CheckDataRegion(const TBlob &src, const TBlob &dst) {
   EXPECT_EQ(equals, 0);
 }
 
-inline unsigned gen_rand_seed() {
-  time_t timer;
-  ::time(&timer);
-  return static_cast<unsigned>(timer);
-}
-
 inline float RandFloat() {
-  static unsigned seed = gen_rand_seed();
-  double v = rand_r(&seed) * 1.0 / RAND_MAX;
-  return static_cast<float>(v);
+  static thread_local std::random_device device;
```

Comment:
This PR is getting huge, I don't understand all the different changes and how the relate to one another. I very much would like to have the CMake changes merged, I think it would help you and reviewers if you could break down the changes in smaller, more manageable independent changes. Otherwise please add notes on why is best that the changes go together.

Reply:
Yes, the change is big. But with a few exceptions all of the changes are closely related. It is extremely hard to satisfy all checks on the CI with CMake changes. Potentially I could take out the openblas source build and android related rand_s changes to a separate PR. Would that be an option?